### PR TITLE
Add 1.28 to release schedule for 1.28 release

### PR DESCRIPTION
**Pull request by Rishit-dagli**
This PR adds 1.28 to the releases schedule for the 1.28.0 release
/milestone 1.28

### CODE REVIEW

File: data/releases/schedule.yaml [hunk 1]
```diff
@@ -2,6 +2,12 @@
 # This file helps to populate the /releases page, and is also parsed to find out the
 # latest patch version for a minor release.
 schedules:
+- release: 1.28
+  releaseDate: 2023-08-15
+  next: 1.28.1
```
Comment (sftim) [inline data/releases/schedule.yaml:7]:
```suggestion
  next:
   release: 1.28.1
```

File: data/releases/schedule.yaml [hunk 2]
```diff
@@ -2,6 +2,12 @@
...
+  cherryPickDeadline: 2023-09-08
+  targetDate: 2023-09-13
+  endOfLifeDate: 2024-10-28
```
Comment (sftim) [inline data/releases/schedule.yaml:10]:
```suggestion
  endOfLifeDate: 2024-10-28
  previousPatches:
    - release: 1.28.0
      targetDate: 2023-08-15
```

**Review by sftim — COMMENTED** [review]
/hold

OK to unhold if either:
- the date of release is 100% confirmed
- the date of release is not confirmed, but SIG Release are tracking any rework on this change if the release happens on a different date

File: data/releases/schedule.yaml (hunk 2 above)
Comment (Rishit-dagli) [reply data/releases/schedule.yaml:11]: Ah yes, thanks, maybe this also needs a change in the docs lead handbook, should I go ahead with doing that too?

**Comment by sftim** [comment]
The build error:
```
Error building site: /opt/build/repo/content/en/releases/patch-releases.md:88:1: failed to render shortcode release-branches: failed to process shortcode: /opt/build/repo/layouts/shortcodes/release-branches.html:16:17: execute of template failed: template: shortcodes/release-branches.html:16:17: executing shortcodes/release-branches.html at <time.Format>: error calling Format: unable to cast <nil> of type <nil> to Time
```

**Review by sftim — COMMENTED** [review]
We might need to set `maintenanceModeStartDate` (_or_ change the shortcode to allow that field to be optional)

**Comment by sftim** [comment]
No direct change to docs. Needs review from SIG Release.

**Comment by Rishit-dagli** [comment]
We will merge this tomorrow using write access.

**Pull request merged**